main.py

```diff
@@ -299,19 +315,18 @@ def _exit():
 
     # Important to stop all docker images upon exit
     # Also make sure that none of those images are running at this point
+    cleanup = build_cleanup(settings)
     atexit.register(cleanup)
     cleanup()
```

Comment:
Why do we call cleanup just after we register it in the atexit? Won't it be called twice?